sunspot_rails/lib/sunspot/rails.rb

```diff
@@ -42,28 +42,44 @@ def build_session(configuration = self.configuration)
       private
 
       def master_config(sunspot_rails_configuration)
-        config = Sunspot::Configuration.build
-        builder = sunspot_rails_configuration.scheme == 'http' ? URI::HTTP : URI::HTTPS
-        config.solr.url = builder.build(
-          :host => sunspot_rails_configuration.master_hostname,
-          :port => sunspot_rails_configuration.master_port,
-          :path => sunspot_rails_configuration.master_path,
-          :userinfo => sunspot_rails_configuration.userinfo
-        ).to_s
-        config.solr.read_timeout = sunspot_rails_configuration.read_timeout
-        config.solr.open_timeout = sunspot_rails_configuration.open_timeout
-        config
+        build_config(sunspot_rails_configuration) do |config|
+          config.solr.url = if sunspot_rails_configuration.master_hosts.empty?
+            builder = sunspot_rails_configuration.scheme == 'http' ? URI::HTTP : URI::HTTPS
+
+            builder.build(
+              :host => sunspot_rails_configuration.master_hostname,
+              :port => sunspot_rails_configuration.master_port,
+              :path => sunspot_rails_configuration.master_path,
+              :userinfo => sunspot_rails_configuration.userinfo
+            ).to_s
+          else
+            sunspot_rails_configuration.hosts
+          end
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [`end` at 57, 10 is not aligned with `if` at 46, 28.](https://www.codacy.com/app/joni-amir/sunspot/file/2482716221/issues/source?bid=3265104&fileBranchId=3265130#l57)